### PR TITLE
Completed sgt-back-end exercise.

### PR DESCRIPTION
**Pull request by rachel-lyy**
POST
sgt-back-end-400-course
sgt-back-end-400-name
sgt-back-end-400-score
sgt-back-end-POST-201
sgt-back-end-POST-500
DELETE 
sgt-back-end-DELETE-204
sgt-back-end-DELETE-400
sgt-back-end-DELETE-404
sgt-back-end-DELETE-500
GET
sgt-back-end-GET-200
sgt-back-end-GET-500
PUT
sgt-back-end-PUT-200
sgt-back-end-PUT-400-gradeId
sgt-back-end-PUT-400-score
sgt-back-end-PUT-404
sgt-back-end-PUT-404-name
sgt-back-end-PUT-500
sgt-back-end-PUT-course

### CODE REVIEW

File: sgt-back-end/get.js
Comment (TimelikeClosure) [inline sgt-back-end/get.js:1]: Please double-check that you're not adding files that are going to remain empty.

File: sgt-back-end/server.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    // check for missing values
+    const { name, course } = req.body;
+    const score = Number(req.body.score);
+    // check values
+    if (!name) {
+      res.status(400).json({ error: 'name must be provided' });
+    } else if (!course) {
+      res.status(400).json({ error: 'course must be provided' });
+    } else if (!score || score <= 0 || score > 100) {
+      res.status(400).json({ error: 'score must be a positive integer' });
```
Comment (TimelikeClosure) [inline sgt-back-end/server.js]: A `score` of `0` should not cause a `400 Bad Request` response:

---

image

File: sgt-back-end/server.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    const score = Number(req.body.score);
+    const gradeId = Number(req.params.gradeId);
+    if (!Number.isInteger(gradeId) || gradeId < 0) {
+      res.status(400).json({ error: '"gradeId" must be a positive integer' });
+    } else if (!name) {
+      res.status(400).json({ error: 'name must be provided' });
+    } else if (!course) {
+      res.status(400).json({ error: 'course must be provided' });
+    } else if (score < 0 || score > 100) {
+      res.status(400).json({ error: 'score must be a positive integer' });
```
Comment (TimelikeClosure) [inline sgt-back-end/server.js]: If `score` is a positive non-integer, it should still cause a `400 Bad Request`: https://en.wikipedia.org/wiki/Integer

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by rachel-lyy** [comment]
**POST**
sgt-back-end-201-POST-update-success
sgt-back-end-400-POST-update-negInt
sgt-back-end-400-POST-update-nonNum
**PUT** 
sgt-back-end-400-PUT-update-negInt
sgt-back-end-400-PUT-update-nonNum

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**